packages/client/tests/functional/chunking-query/tests.ts

```diff
@@ -0,0 +1,263 @@
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient, Tag } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+testMatrix.setupTestSuite(
+  ({ provider, providerFlavor }) => {
+    function generatedIds(n: number) {
+      // ["1","2",...,"n"]
+      const ids = Array.from({ length: n }, (_, i) => i + 1)
+      return ids
+    }
+
+    describe('issues #8832 / #9326 success cases', () => {
+      async function createTags(n: number): Promise<number[]> {
+        const ids = generatedIds(n)
+        const data = ids.map((id) => ({ id }))
+        await prisma.tag.createMany({
+          data,
+        })
+        return ids
+      }
+
+      async function getTagsFindManyInclude(): Promise<Tag[]> {
+        const tags = await prisma.tag.findMany({
+          include: {
+            posts: true,
+          },
+        })
+        return tags
+      }
+
+      async function getTagsParams(ids: number[]): Promise<Tag[]> {
+        const idsParams = ids.map((paramIdx) => {
+          const param = ['mysql'].includes(provider) ? '?' : `\$${paramIdx}`
+          return param
+        })
+
+        const tags = await prisma.$queryRawUnsafe<Tag[]>(
+          `
+            SELECT *
+            FROM tag
+            WHERE id IN (${idsParams.join(', ')})
+          `,
+          ...ids,
+        )
+        return tags
+      }
+
+      async function getTagsFindManyIn(ids: number[]): Promise<Tag[]> {
+        const tags = await prisma.tag.findMany({
+          where: {
+            id: {
+              in: ids,
+            },
+          },
+        })
+        return tags
+      }
+
+      async function clean() {
+        const cleanPrismaPromises = [prisma.tagsOnPosts.deleteMany(), prisma.post.deleteMany(), prisma.tag.deleteMany()]
+        for (const promise of cleanPrismaPromises) {
+          await promise
+        }
```

Comment:
Nit: I'm more used to write a transaction for this, because the order is guaranteed and it makes the code easier I think like
```suggestion
        await prisma.$transaction([
          prisma.tagsOnPosts.deleteMany(), 
          prisma.post.deleteMany(), 
          prisma.tag.deleteMany()
        ])
```